### PR TITLE
*: add a column describing memory usage for table information_schema.processlist

### LINKED ISSUE #10199: support "SHOW PROCESSLIST MEMINFO"
## Feature Request

**Is your feature request related to a problem? Please describe:**

In https://github.com/pingcap/tidb/pull/8543, we removed the memory info from the result of `show processlist`. As discussed in https://github.com/pingcap/tidb/pull/8543#issuecomment-444119260, we decide to add a new command like `show processlist meminfo` to show the process lists + meminfo

**Describe the feature you'd like:**

use `show processlist meminfo` to show the result of process lists and meminfo

**Describe alternatives you've considered:**

No

**Teachability, Documentation, Adoption, Migration Strategy:**

No

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

Closes #10199

### What is changed and how it works?

Add a column describing memory usage for table `information_schema.processlist` and keep `show processlist` acting as before.

```mysql
mysql> SELECT * FROM information_schema.processlist;
+------+------+-----------+------+---------+------+-------+----------------------------------------------+------+
| ID   | USER | HOST      | DB   | COMMAND | TIME | STATE | INFO                                         | MEM  |
+------+------+-----------+------+---------+------+-------+----------------------------------------------+------+
|    1 | root | 127.0.0.1 | test | Query   |    1 | 2     | select * from t, ta                          |  856 |
|    2 | root | 127.0.0.1 | test | Query   |    0 | 2     | SELECT * FROM information_schema.processlist |    0 |
+------+------+-----------+------+---------+------+-------+----------------------------------------------+------+
2 rows in set (0.00 sec)

mysql> show processlist;
+------+------+-----------+------+---------+------+-------+----------------------+
| Id   | User | Host      | db   | Command | Time | State | Info                 |
+------+------+-----------+------+---------+------+-------+----------------------+
|    2 | root | 127.0.0.1 | test | Query   |    0 | 2     | show processlist     |
|    1 | root | 127.0.0.1 | test | Query   |    1 | 2     | select *  from t, ta |
+------+------+-----------+------+---------+------+-------+----------------------+
2 rows in set (0.01 sec)
```

### Check List 

Tests 

 - Unit test
 - Manual test (see above)

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - Need to update the documentation
 - Need to be included in the release note?

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/10837)

### CODE REVIEW

**Comment by SunRunAway** [comment]
@zz-jason @XuHuaiyu PTAL

**Comment by SunRunAway** [comment]
@XuHuaiyu sorry, PTAL again.

File: util/misc_test.go [hunk 1]
```diff
@@ -152,9 +155,12 @@ func (s *testMiscSuite) TestBasicFunc(c *C) {
 		Time:    time.Now(),
 		State:   1,
 		Info:    "test",
+		StmtCtx: &stmtctx.StatementContext{
+			MemTracker: memory.NewTracker(stringutil.StringerStr(""), -1),
+		},
 	}
-	row := pi.ToRow(false)
-	row2 := pi.ToRow(true)
+	row := pi.ToRowForShow(false)
```
Comment (zz-jason) [inline util/misc_test.go]: Can we rewrite this test to not call `pi.ToRowForShow(false)`? Such that we can remove the parameter `full` in `ToRowForShow(full bool)`

Comment (SunRunAway) [reply util/misc_test.go]: `SHOW FULL PROCESSLIST` calls `pi.ToRowForShow(true)` and `SHOW PROCESSLIST` calls `pi.ToRowForShow(false)`.

**Pull request merged**